src/DynamoCore/Engine/EngineController.cs

```diff
@@ -147,6 +147,7 @@ public EngineController(LibraryServices libraryServices, string geometryFactoryF
             this.libraryServices = libraryServices;
             libraryServices.LibraryLoaded += LibraryLoaded;
             CompilationServices = new CompilationServices(libraryServices);
+            codeCompletionServices = new CodeCompletionServices(libraryServices.LibraryManagementCore);
```

Comment:
Initialize `CodeCompletionServices` with `LibraryManagementCore` and not `LiveRunnerCore` as was done previously. There are subtle differences between these 2 cores, and they are not always in sync. For example, `LibraryManagementCore` is used for library management, keeps compile-time records of libraries loaded into the system, the information used to display nodes in the Library, for code block precompilation and for CBN autocomplete whereas `LiveRunnerCore` has runtime information and is used to construct the runtime Core of the VM.

Reply:
Will the LibraryManagmentCore be missing any functions, like those defined in code blocks?

Reply:
No. It will include those.

Reply:
When a codeblock node is edited, any new code is precompiled. This is done using the library management (compile-time) core. That is when new DS function definitions if any are added, which are then available for use in auto-completion.